evap/staff/tests/test_views.py

```diff
@@ -2389,6 +2389,23 @@ def test_copy_participants(self):
             self.evaluation.participants.count(), original_participant_count + self.evaluation2.participants.count()
         )
 
+    def test_copy_invalid_participants(self):
+        page = self.app.get(self.url, user=self.manager)
+
+        old_evaluation = baker.make(
+            Evaluation,
+            course__semester=self.evaluation2.course.semester,
+            participants=self.evaluation2.participants.all(),
+        )
+        old_pk = old_evaluation.pk
+        old_evaluation.delete()
+
+        form = page.forms["participant-copy-form"]
+        form["pc-evaluation"] = str(self.evaluation2.pk)
+        params = dict(form.submit_fields() + [("operation", "copy-replace-participants")])
+        params["pc-evaluation"] = str(old_pk)
+        form.response.goto(form.action, method=form.method, params=params, status=400)
```

Comment:
I can't do that, I then get `ValueError("Option '7' not found (from '', '5', '6')")` while setting the value :( 
I somewhat copied this code from `submit_with_modal` but don't know a better solution

Reply:
The `submit` call checks that you only submit what is possible from the HTML - if there is no submit button with the given value, it will complain

For modals, I used this code because a modal with name and value attributes behaves like a submit button, but webtest does not know our modals of course

I havent read the diff here closely enough yet to judge whether the usage of this trick is appropriate here

Reply:
I would have expected everything to work here since we're `self.app.get`ting the page before we delete the `old_evaluation`, so for webtest, the PK should still be a valid option. Will have to look into this.

Reply:
you are right! I just get the page before creating the evaluation. I'll update this accordingly.